### ISSUE TITLE
False Negative: NP_LOAD_OF_KNOWN_NULL_VALUE missing when null check is done via instanceof

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3916]
**Describe the bug**
The rule **NP_LOAD_OF_KNOWN_NULL_VALUE** fails to report null pointer dereference when the null check is performed using instanceof instead of an explicit != null comparison. 

**To Reproduce**
```java
class Foo {
  private void foo1(String s1) {
      if (s1 != null) return;
      String str = s1.toString();  // <- True Positive: NP reported
  }
  
  private void foo2(String s2) {
      if (s2 instanceof String) return;
      String str = s2.toString();  // <- False Negative: NP not reported
  }
}
```

**Expected behavior**
Both foo1() and foo2() should be reported with the NP warning.

**Actual behavior**
Only foo1() is reported with NP . foo2() is silently ignored.

**Environment**
- Java version: JDK21
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI
   

**Additional context**
When a parameter with static type T is checked using instanceof T, 
this is semantically equivalent to a null check:
```java
// When x has static type T:
x instanceof T ≡ x != null
```
However, SpotBugs only detects the explicit null check pattern and misses the equivalent instanceof pattern.

**Comment by leemeii** [comment]
Hi! I wrote a quick program to verify that these two forms are indeed equivalent.

Tested both methods with `null` input to confirm this is a genuine bug:
```java
public class Test {
    static void foo1(String s1) {
        if (s1 != null) return;
        s1.toString();
    }
    
    static void foo2(String s2) {
        if (s2 instanceof String) return;
        s2.toString();
    }
    
    public static void main(String[] args) {
        try { foo1(null); } catch (NullPointerException e) { System.out.println("foo1 NPE!"); }
        try { foo2(null); } catch (NullPointerException e) { System.out.println("foo2 NPE!"); }
    }
}
```

**Result:** Both throw NPE!
```
foo1 NPE!
foo2 NPE!
```

Both methods have **identical runtime behavior**, yet Spotbugs only detects `foo1()` and misses `foo2()`.
